pkg/oci/remote.go

```diff
@@ -14,8 +16,52 @@ var (
 	ErrNotAnOCIImage = errors.New("not an OCI image")
 )
 
+type configs struct {
+	Auths map[string]struct {
+		Username string `json:"username"`
+		Password string `json:"password"`
+		Email    string `json:"email"`
+		Auth     string `json:"auth"`
+	} `json:"auths"`
+}
+
 func getRepository(reference string) (*remote.Repository, error) {
 	repos := make(map[string]auth.Credential)
+
+	homedir, err := os.UserHomeDir()
+	if err != nil {
+		return nil, err
+	}
+
+	configFile := filepath.Join(homedir, ".docker/config.json")
+	if _, err := os.Stat(configFile); err == nil {
+		contents, err := os.ReadFile(configFile)
+		if err != nil {
+			fmt.Println("Error reading config file:", err)
+			return nil, err
+		}
+
+		var config configs
+		if err := json.Unmarshal(contents, &config); err != nil {
+			fmt.Println("Error parsing config file:", err)
+			return nil, err
+		}
+
+		for registry, authconfigs := range config.Auths {
+			if authconfigs.Username != "" && authconfigs.Password != "" {
+				fmt.Println("Adding registry credentials from file: ", registry)
+				repos[registry] = auth.Credential{
+					Username: authconfigs.Username,
+					Password: authconfigs.Password,
+				}
+			}
+		}
+	} else if os.IsNotExist(err) {
+		fmt.Println(configFile, "file does not exist.")
+	} else {
+		fmt.Println("Error checking config file:", err)
```

Comment:
This error message is to vague in that the user does not know what config file its referring to